File: recursion/sum_of_numbers.js
/**
 * @param {number} n
 * @returns {number}
 */
function sum_of_numbers_iterative(n) {
	let sum = 0;
	for (let i = 0; i <= n; i++) {
		sum += i;
	}
	return sum;
}

/**
 * @param {number} n
 * @returns {number}
 */
function sum_of_numbers_recursive(n) {
	if (n == 0)
		return 0;
	return n + sum_of_numbers_recursive(n - 1);
}

function run(n) {
	console.log(`${n} - sum = ${sum_of_numbers_iterative(n)}`);
	console.log(`${n} - sum = ${sum_of_numbers_recursive(n)}`);
	console.log();
}

function main() {
	console.log();
	run(3);
	run(10);
	run(100);
	run(1000);
}

main();

File: recursion/sum_of_numbers.c
#include <stdio.h>

int	sum_of_numbers_iterative(int n)
{
	int	sum;
	int	i;

	sum = 0;
	i = 0;
	while (i < n)
		sum += i++ + 1;
	return (sum);
}

int	sum_of_numbers_recursive(int n)
{
	if (n == 0)
		return (0);
	return (n + sum_of_numbers_recursive(n - 1));
}

int	main(void)
{
	int	n;

	n = 10;
	printf("%d\tsum = %d\n", n, sum_of_numbers_iterative(n));
	printf("%d\tsum = %d\n", n, sum_of_numbers_recursive(n));

	n = 100;
	printf("%d\tsum = %d\n", n, sum_of_numbers_iterative(n));
	printf("%d\tsum = %d\n", n, sum_of_numbers_recursive(n));

	n = 1000;
	printf("%d\tsum = %d\n", n, sum_of_numbers_iterative(n));
	printf("%d\tsum = %d\n", n, sum_of_numbers_recursive(n));
	return (0);
}
